libqpdf/qpdf-c.cc

```diff
@@ -1284,6 +1340,19 @@ char const* qpdf_oh_get_real_value(qpdf_data qpdf, qpdf_oh oh)
         });
 }
 
+QPDF_BOOL qpdf_oh_get_value_as_real(
+    qpdf_data qpdf, qpdf_oh oh, char const** value, size_t* length)
+{
+    return do_with_oh<QPDF_BOOL>(
+        qpdf, oh, return_false, [qpdf, value, length](QPDFObjectHandle& o) {
+            QTC::TC("qpdf", "qpdf-c called qpdf_oh_get_value_as_real");
+            auto result = o.getValueAsReal(qpdf->tmp_string);
+            *value = qpdf->tmp_string.c_str();
```

Comment:
wouldn't it be better to only do this if result is true?

Reply:
Not only better, but also correct. Changed